goforit.go

```diff
@@ -424,5 +441,20 @@ func (g *goforit) Close() error {
 	return nil
 }
 
+func (g *goforit) ReportCounts(callback func(name string, total, enabled uint64)) {
+	g.reportMu.Lock()
+	defer g.reportMu.Unlock()
+
+	for name, fh := range g.flags.load() {
```

Comment:
We can have rather a lot of flags to iterate over! I guess it's still preferable to move this out of the synchronous Enabled calls, but it could be a lot of overhead for services that use flags rarely.

Reply:
@vasi-stripe yeah - I think it should be fine, as we're walking a map without a lock that affects Enabled held, but I added a metric here so that we can see how long this actually takes in QA and prod in a few specific canary services.